### PR TITLE
Lower CPU usage by doing less display updates

### PR DESCRIPTION
**Pull request by mikaoj**
**TL;DR:** Lowers CPU usage from 40-50% to 10-15% while recording.

Currently Audacity consumes ~90% CPU when recording on my 2019 Macbook Pro with 2,3 GHz 8-Core Intel Core i9 running macOS 11.6. That will kick the fans into jet engine mode which will be audible if you have the computer in the same room.
One trick is to toggle off "update display while playing" but it will still use 40-50%.

By removing this code (and having update display while playing toggled off) I'm able to get it to 10-15%.
I haven't noticed any artifacts or components not updating correctly, so to me this forced update seems to be done needlessly. But I'm expecting to be told otherwise in this PR :)

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by crsib** [comment]
Thank you a lot for your contribution! We have plans to fix the rendering performance, so we can keep the visual feedback from the recording (and make it smooth!) for 3.2

**Comment by mikaoj** [comment]
> so we can keep the visual feedback from the recording

Out of curiosity so I learn something about the code base. Which visual feedback did this PR break? :)

**Comment by crsib** [comment]
Well, waveform wasn't redrawn while recording unless the horizontal scrolling started :)
